crates/database/src/ethflow_orders.rs

```diff
@@ -46,12 +46,14 @@ pub async fn read_order(
 }
 
 pub async fn mark_eth_orders_as_refunded(
-    ex: &mut PgTransaction<'_>,
+    db: &mut PgConnection,
     uids: &[OrderUid],
 ) -> Result<(), sqlx::Error> {
+    let mut transaction = db.begin().await?;
     for uid in uids.iter() {
-        mark_eth_order_as_refunded(ex, uid).await?;
+        mark_eth_order_as_refunded(&mut transaction, uid).await?;
     }
+    transaction.commit().await?;
     Ok(())
 }
```

Comment:
The way I intended for these functions to work is that you don't create transactions inside of them. See the comment in database/src/lib.rs .  I prefer creating the transaction outside.